stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -1599,17 +1608,26 @@ extension ${Self} {
   /// can't be represented exactly, the result is `nil`.
   ///
   /// - Parameter value: The integer to represent as a floating-point value.
-%   if srcBits < SignificandBitCount:
+%   if srcBits <= dstBits:
```

Comment:
This change doesn't seem right; at least, it's not behavior preserving even for unsigned values. `srcBits <= dstBits` is equivalent to `srcBits <= SignificandBitCount + 1`, which is equivalent to `srcBits - 1 <= SignificandBitCount`, which isn't the same as `srcBits < SignificandBitCount`.

Reply:
Yes, this is deliberate. The old code was doing unnecessary checks. By making this condition correct we can simplify the exactness checking code (by avoiding the possible overflow).

For example, the maximum value that can be represented with an unsigned 24-bit integer (or a signed 25-bit integer) is 16777215 (0xffffff). This value can be encoded exactly by a Float32 which has a 23-bit significand (24-bit including the implicit bit). Therefore we can represent any possible value of an unsigned integer exactly if the length of the integer is less than or equal to the number of significand bits plus one. Assuming the exponent is large enough of course.